src/main/scala/com/cognite/spark/datasource/SdkV1Rdd.scala

```diff
@@ -0,0 +1,41 @@
+package com.cognite.spark.datasource
+
+import cats.effect.IO
+import com.cognite.sdk.scala.v1.GenericClient
+import com.cognite.sdk.scala.common.Readable
+import org.apache.spark.{Partition, SparkContext, TaskContext}
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.Row
+
+case class SdkPartition(cursor: Option[String], size: Option[Int], index: Int) extends Partition
+
+case class SdkV1Rdd[A](
+    @transient override val sparkContext: SparkContext,
+    config: RelationConfig,
+    cursors: Iterator[(Option[String], Option[Int])],
+    toRow: A => Row,
+    clientToResource: GenericClient[IO, Nothing] => Readable[A, IO])
+    extends RDD[Row](sparkContext, Nil) {
+  import CdpConnector.sttpBackend
+  override def getPartitions: Array[Partition] =
+    cursors.toIndexedSeq.zipWithIndex.map {
+      case ((cursor, size), index) =>
+        val partitionSize = (config.limit, size) match {
+          case (None, s) => s
+          case (Some(l), Some(s)) => Some(scala.math.min(l, s))
+          case (l, None) => l
+        }
+        SdkPartition(cursor, partitionSize, index)
+    }.toArray
+  override def compute(_split: Partition, context: TaskContext): Iterator[Row] = {
+    val split = _split.asInstanceOf[SdkPartition]
+    implicit val auth = config.auth
+    val client = new GenericClient[IO, Nothing]("cdp-spark-datasource")
```

Comment:
We should include the Spark data source version in the application name to be able to track usage for different releases.

Reply:
Good point, I'll use `Constants.SparkDatasourceVersion` to match with `x-cdp-sdk`.